File: Tugas-PostDay4/src/demo_package/src/serverNode.cpp
#include "ros/ros.h"
#include "demo_package/MagnitudeCalculation.h"
#include <cmath>

bool magnitudeHitung(demo_package::MagnitudeCalculation::Request &req, demo_package::MagnitudeCalculation::Response &resp){
    resp.magnitude = std::sqrt(req.komponenX * req.komponenX + req.komponenY * req.komponenY + req.komponenZ * req.komponenZ);

    return true;
}

int main(int argc, char**argv){
    // Inisialisasi Node
    ros::init(argc,argv,"server_node");

    // Inisialisasi Handler Node
    ros::NodeHandle nh;

    // Inisialisasi Server dari Handler
    ros::ServiceServer server = nh.advertiseService("MagnitudeCalculation",magnitudeHitung);
    ROS_INFO("Siap menghitung magnitude capt!");
    ros::spin();

    return 0;
}

File: Tugas-PostDay4/src/demo_package/script/ServerNode.py
#!/usr/bin/env python3

import rospy
from demo_package.srv import MagnitudeCalculation, MagnitudeCalculationResponse
from math import sqrt

def hitung_magnitude(req):
    magnitude = sqrt(req.komponenX**2 + req.komponenY**2 + req.komponenZ**2)

    rospy.loginfo("Menerima ingfo: X=%f, Y=%f, Z=%f", req.komponenX, req.komponenY, req.komponenZ)
    rospy.loginfo("Mengirim ingfo...")

    return MagnitudeCalculationResponse(magnitude)

def magnitude_server():
    rospy.init_node('server_node')
    s = rospy.Service('calculate_magnitude', MagnitudeCalculation, hitung_magnitude)
    rospy.loginfo("Siap menghitung magnitude.")
    rospy.spin()

if __name__ == "__main__":
    magnitude_server()
